// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 3 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet  restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de
// command took 3.76 sec and exited with 0
// start dotnet  build -c Release --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de
// command took 5.27 sec and exited with 0
// ***** Done, took 00:00:09 (9.19 sec)   *****
// Found 3 benchmarks:
//   BenchmarkDriver.NpoiTest: DefaultJob
//   BenchmarkDriver.EPPlusTest: DefaultJob
//   BenchmarkDriver.ClosedXmlTest: DefaultJob

Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)
// **************************
// Benchmark: BenchmarkDriver.NpoiTest: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet 5ae95308-2782-4186-bc23-b363f44e67de.dll --anonymousPipes 1564 1504 --benchmarkName BenchmarkDriver.NpoiTest --job Default --benchmarkId 0 in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\bin\Release\net6.0
// BeforeAnythingElse

// Benchmark Process Environment Information:
// BenchmarkDotNet v0.13.7
// Runtime=.NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2
// GC=Concurrent Workstation
// HardwareIntrinsics=AVX2,AES,BMI1,BMI2,FMA,LZCNT,PCLMUL,POPCNT VectorSize=256
// Job: DefaultJob

OverheadJitting  1: 1 op, 292400.00 ns, 292.4000 us/op

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ObjectDisposedException: GetRecords<T>() returns an IEnumerable<T> that yields records. This means that the method isn't actually called until you try and access the values. e.g. .ToList() Did you create CsvReader inside a using block and are now trying to access the records outside of that using block?
Object name: 'CsvReader'.
   at CsvHelper.CsvReader.GetRecords[T]()+MoveNext()
   at BenchmarkDriver.CreateDataTableFromIEnumerable[T](IEnumerable`1 data) in C:\source\repos\excel-benchmarking\Benchmarking\Program.cs:line 128
   at BenchmarkDriver.NpoiTest() in C:\source\repos\excel-benchmarking\Benchmarking\Program.cs:line 34
   at BenchmarkDotNet.Autogenerated.Runnable_0.WorkloadActionNoUnroll(Int64 invokeCount) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 311
   at BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   at BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   at BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   at BenchmarkDotNet.Autogenerated.Runnable_0.Run(IHost host, String benchmarkName) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 176
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 57
// AfterAll
No Workload Results were obtained from the run.
// Benchmark Process 70744 has exited with code -1.

// ** Remained 2 (66.7%) benchmark(s) to run. Estimated finish 2023-08-24 8:51 (0h 0m from now) **
Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)
// **************************
// Benchmark: BenchmarkDriver.EPPlusTest: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet 5ae95308-2782-4186-bc23-b363f44e67de.dll --anonymousPipes 1460 1544 --benchmarkName BenchmarkDriver.EPPlusTest --job Default --benchmarkId 1 in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\bin\Release\net6.0
// BeforeAnythingElse

// Benchmark Process Environment Information:
// BenchmarkDotNet v0.13.7
// Runtime=.NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2
// GC=Concurrent Workstation
// HardwareIntrinsics=AVX2,AES,BMI1,BMI2,FMA,LZCNT,PCLMUL,POPCNT VectorSize=256
// Job: DefaultJob

OverheadJitting  1: 1 op, 295900.00 ns, 295.9000 us/op

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ObjectDisposedException: GetRecords<T>() returns an IEnumerable<T> that yields records. This means that the method isn't actually called until you try and access the values. e.g. .ToList() Did you create CsvReader inside a using block and are now trying to access the records outside of that using block?
Object name: 'CsvReader'.
   at CsvHelper.CsvReader.GetRecords[T]()+MoveNext()
   at BenchmarkDriver.EPPlusTest() in C:\source\repos\excel-benchmarking\Benchmarking\Program.cs:line 63
   at BenchmarkDotNet.Autogenerated.Runnable_1.WorkloadActionNoUnroll(Int64 invokeCount) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 503
   at BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   at BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   at BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   at BenchmarkDotNet.Autogenerated.Runnable_1.Run(IHost host, String benchmarkName) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 368
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 57
// AfterAll
No Workload Results were obtained from the run.
// Benchmark Process 14656 has exited with code -1.

// ** Remained 1 (33.3%) benchmark(s) to run. Estimated finish 2023-08-24 8:51 (0h 0m from now) **
Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: High performance)
// **************************
// Benchmark: BenchmarkDriver.ClosedXmlTest: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet 5ae95308-2782-4186-bc23-b363f44e67de.dll --anonymousPipes 1504 1484 --benchmarkName BenchmarkDriver.ClosedXmlTest --job Default --benchmarkId 2 in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\bin\Release\net6.0
// BeforeAnythingElse

// Benchmark Process Environment Information:
// BenchmarkDotNet v0.13.7
// Runtime=.NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2
// GC=Concurrent Workstation
// HardwareIntrinsics=AVX2,AES,BMI1,BMI2,FMA,LZCNT,PCLMUL,POPCNT VectorSize=256
// Job: DefaultJob

OverheadJitting  1: 1 op, 353900.00 ns, 353.9000 us/op

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ObjectDisposedException: GetRecords<T>() returns an IEnumerable<T> that yields records. This means that the method isn't actually called until you try and access the values. e.g. .ToList() Did you create CsvReader inside a using block and are now trying to access the records outside of that using block?
Object name: 'CsvReader'.
   at CsvHelper.CsvReader.GetRecords[T]()+MoveNext()
   at BenchmarkDriver.ClosedXmlTest() in C:\source\repos\excel-benchmarking\Benchmarking\Program.cs:line 99
   at BenchmarkDotNet.Autogenerated.Runnable_2.WorkloadActionNoUnroll(Int64 invokeCount) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 695
   at BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   at BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   at BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   at BenchmarkDotNet.Autogenerated.Runnable_2.Run(IHost host, String benchmarkName) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 560
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\source\repos\excel-benchmarking\Benchmarking\bin\Release\net6.0\5ae95308-2782-4186-bc23-b363f44e67de\5ae95308-2782-4186-bc23-b363f44e67de.notcs:line 57
// AfterAll
No Workload Results were obtained from the run.
// Benchmark Process 75396 has exited with code -1.

// ** Remained 0 (0.0%) benchmark(s) to run. Estimated finish 2023-08-24 8:51 (0h 0m from now) **
Successfully reverted power plan (GUID: 381b4222-f694-41f0-9685-ff5bb260df2e FriendlyName: Balanced)
// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts\results\BenchmarkDriver-report.csv
  BenchmarkDotNet.Artifacts\results\BenchmarkDriver-report-github.md
  BenchmarkDotNet.Artifacts\results\BenchmarkDriver-report.html

// * Detailed results *
BenchmarkDriver.NpoiTest: DefaultJob
Runtime = .NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2; GC = Concurrent Workstation
There are not any results runs

BenchmarkDriver.EPPlusTest: DefaultJob
Runtime = .NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2; GC = Concurrent Workstation
There are not any results runs

BenchmarkDriver.ClosedXmlTest: DefaultJob
Runtime = .NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2; GC = Concurrent Workstation
There are not any results runs

// * Summary *

BenchmarkDotNet v0.13.7, Windows 11 (10.0.22621.2134/22H2/2022Update/SunValley2)
Intel Core i9-9880H CPU 2.30GHz, 1 CPU, 16 logical and 8 physical cores
.NET SDK 6.0.100
  [Host]     : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2
  DefaultJob : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT AVX2


|        Method | Mean | Error |
|-------------- |-----:|------:|
|      NpoiTest |   NA |    NA |
|    EPPlusTest |   NA |    NA |
| ClosedXmlTest |   NA |    NA |

Benchmarks with issues:
  BenchmarkDriver.NpoiTest: DefaultJob
  BenchmarkDriver.EPPlusTest: DefaultJob
  BenchmarkDriver.ClosedXmlTest: DefaultJob

// * Warnings *
Environment
  Summary -> Detected error exit code from one of the benchmarks. It might be caused by following antivirus software:
        - Sophos Intercept X (C:\Program Files\Sophos\Endpoint Defense\SEDcli.exe)
        - Sophos Anti-Virus (C:\Program Files (x86)\Sophos\Sophos Anti-Virus\WSCClient.exe)
        - Windows Defender (windowsdefender://)
        - Sophos Intercept X (C:\Program Files\Sophos\Endpoint Defense\SEDcli.exe)
        - Sophos Intercept X (C:\Program Files\Sophos\Endpoint Defense\SEDcli.exe)
Use InProcessEmitToolchain or InProcessNoEmitToolchain to avoid new process creation.


// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ns  : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
Run time: 00:00:03 (3.04 sec), executed benchmarks: 3

Global total time: 00:00:13 (13.08 sec), executed benchmarks: 3
// * Artifacts cleanup *
Artifacts cleanup is finished
